AutoCollection/HttpRequests.ts

```diff
@@ -75,9 +75,38 @@ class AutoCollectHttpRequests {
         );
     }
 
+    private _registerRequest(request: http.ServerRequest, response: http.ServerResponse, onRequest: Function) {
+        // Set up correlation context
+        const requestParser = new HttpRequestParser(request);
+        const correlationContext = this._generateCorrelationContext(requestParser);
+
+        // Note: Check for if correlation is enabled happens within this method.
+        // If not enabled, function will directly call the callback.
+        CorrelationContextManager.runWithContext(correlationContext, () => {
+            if (this._isEnabled) {
+                // Mark as auto collected
+                (<any>request)[AutoCollectHttpRequests.alreadyAutoCollectedFlag] = true;
+
+                // Auto collect request
+                AutoCollectHttpRequests.trackRequest(this._client, { request: request, response: response }, requestParser);
+            }
+
+            if (typeof onRequest === "function") {
+                onRequest(request, response);
+            }
+        });
+    }
+
     private _initialize() {
         this._isInitialized = true;
 
+        // Avoid the creation of multiple handler on http(s).createServer
+        if (AutoCollectHttpRequests.HANDLER_READY) {
```

Comment:
You can use _isInitialized , no need to add new flag for the same thing, do not clean it on dispose